File: csharp/src/Ice/ConnectionOptions.cs
// Copyright (c) ZeroC, Inc.

#nullable enable

namespace Ice;

internal sealed record class ConnectionOptions(
    TimeSpan connectTimeout,
    TimeSpan closeTimeout,
    TimeSpan idleTimeout,
    bool enableIdleCheck,
    TimeSpan inactivityTimeout,
    int maxDispatches);

File: js/packages/ice/src/Ice/ConnectionOptions.js
// Copyright (c) ZeroC, Inc.

export class ConnectionOptions {
    constructor(connectTimeout, closeTimeout, idleTimeout, enableIdleCheck, inactivityTimeout) {
        this._connectTimeout = connectTimeout;
        this._closeTimeout = closeTimeout;
        this._idleTimeout = idleTimeout;
        this._enableIdleCheck = enableIdleCheck;
        this._inactivityTimeout = inactivityTimeout;
    }

    get connectTimeout() {
        return this._connectTimeout;
    }

    get closeTimeout() {
        return this._closeTimeout;
    }

    get idleTimeout() {
        return this._idleTimeout;
    }

    get enableIdleCheck() {
        return this._enableIdleCheck;
    }

    get inactivityTimeout() {
        return this._inactivityTimeout;
    }
}
